src/libraries/System.Private.Xml.Linq/tests/Schema/ExtensionTests.cs

```diff
@@ -27,397 +27,405 @@ namespace CoreXml.Test.XLinq
+             * Also the arguments for the folowing tests: XAttributeSuccessValidate
+             * XAttributeFailValidate, XAttributeThrowExceptionValidate, XElementSuccessValidate
+             * XElementFailValidate,XElementThrowExceptionValidate, XAttributeGetSchemaInfo
+             * XElementGetSchemaInfo may need to be changed.
+             */
+            //LoadOutsideDocuments("c:\\note.xsd", "c:\\note.xml");
+        }
+
+        // This gets called when a validation error occurs.
+        private void TestValidationHandler(object sender, ValidationEventArgs e)
+        {
+            validationSucceded = false;
+        }
+
+        // Test succesful validation.
+        [Fact]
+        public void XDocumentSuccessValidate()
+        {
+            validationSucceded = true;
+            ExtensionsClass.Validate(xmlDocument, schemaSet,
+                new ValidationEventHandler(TestValidationHandler));
+            Assert.True(validationSucceded);
+        }
+
+        // Test failed validation.
+        [Fact]
+        public void XDocumentFailValidate()
+        {
+            string elementName = "AlteringElementName";
+            object elementValue = "AlteringElementValue";
+
+            // Alter XML document to invalidate.
+            var newElement = new XElement(elementName, elementValue);
+            xmlDocument.Root.Add(newElement);
+
+            validationSucceded = true;
+            ExtensionsClass.Validate(xmlDocument, schemaSet,
+                new ValidationEventHandler(TestValidationHandler));
+            Assert.False(validationSucceded);
+        }
+
+        /*
+         * test if exception is thrown when xml document does not
+         * conform to the xml schema
+         */
+        [Fact]
+        public void XDocumentThrowExceptionValidate()
+        {
+            string elementName = "AlteringElementName";
+            object elementValue = "AlteringElementValue";
+
+            // Alter XML document to invalidate.
+            var newElement = new XElement(elementName, elementValue);
+            xmlDocument.Root.Add(newElement);
+
+            Assert.Throws<XmlSchemaValidationException>(() => ExtensionsClass.Validate(xmlDocument, schemaSet, null));
+        }
+
+        /*
+         * test xml validation populating the XML tree with
+         * the post-schema-validation infoset(PSVI)
+         */
+        [Fact]
+        public void XDocumentAddSchemaInfoValidate()
+        {
+            // No. of elements before validation.
+            IEnumerable<XElement> elements = xmlDocument.Elements();
+            IEnumerator<XElement> elementsEnumerator = elements.GetEnumerator();
+            int beforeNoOfElements = 0;
+            int beforeNoOfAttributes = 0;
+            while (elementsEnumerator.MoveNext())
+            {
+                    beforeNoOfElements++;
+                    if (!elementsEnumerator.Current.HasAttributes)
+                        continue;
+                    else
```

Comment:
Our style guidelines say that if any branch has braces, all must have braces.  So either the if block should have braces, or the else should be removed (since it's superflous given the continue in the if block).